src/app/core/services/filters.service.ts

```diff
@@ -34,13 +35,47 @@ export const DEFAULT_FILTER: Filter = {
  * Filters are subscribed to and emitted from this service
  */
 export class FiltersService {
+  readonly presetViews: {
+    [key: string]: () => Filter;
+  } = {
+    currentlyActive: () => ({
+      title: '',
+      status: ['open pullrequest', 'merged pullrequest', 'open issue', 'closed issue'],
+      type: 'all',
+      sort: { active: 'status', direction: 'asc' },
+      labels: [],
+      milestones: this.getMilestonesForCurrentlyActive().map((milestone) => milestone.title),
+      hiddenLabels: new Set<string>(),
+      deselectedLabels: new Set<string>()
+    }),
+    contributions: () => ({
+      title: '',
+      status: ['open pullrequest', 'merged pullrequest', 'open issue', 'closed issue'],
+      type: 'all',
+      sort: { active: 'id', direction: 'desc' },
+      labels: [],
+      milestones: this.milestoneService.milestones.map((milestone) => milestone.title),
+      hiddenLabels: new Set<string>(),
+      deselectedLabels: new Set<string>()
+    }),
+    custom: () => this.filter$.value
+  };
+
+  // List of keys in the new filter change that causes current filter to not qualify to be a preset view.
+  readonly presetChangingKeys = new Set<string>(['status', 'type', 'milestones']);
+
   public filter$ = new BehaviorSubject<Filter>(DEFAULT_FILTER);
+  // Either 'currentlyActive', 'contributions', or 'custom'.
+  public presetView$ = new BehaviorSubject<String>('custom');
 
   // Helps in determining whether all milestones were selected from previous repo during sanitization of milestones
   private previousMilestonesLength = 0;
 
+  constructor(private milestoneService: MilestoneService) {}
+
   clearFilters(): void {
     this.filter$.next(DEFAULT_FILTER);
+    this.presetView$.next('custom');
     this.previousMilestonesLength = 0;
   }
```

Comment:
Similar comment as previous one, should we have "Currently Active" as our default view?

Reply:
Fixed as well!